test: Introduce memory tests (#14693)
prisma/prisma: TESTING.md

inserted in its place during the generation. All parameters from the test matrix can be used within that
expression.

## Memory tests

This suite tests client for memory leaks. It works by repeatedly running test code in a loop
and monitoring V8 heap usage after every iteration. If it then detects that memory usage growth

> with a rat, higher then threshold it will report memory leak and file the test.

Reviewer: ```suggestion
with a rate, higher then threshold it will report memory leak and file the test.
```
Nit

Reviewer: What does this means by the way `and file the test`

Author: It means I can't type, it should've been `fail` :D

> with a rate, higher then threshold it will report a memory leak and file the test.

Reviewer: `and file the test.` what does this means?

Reviewer: I saw your comment now
>It means I can't type, it should've been fail :D

Reviewer: ```suggestion
with a rate, higher then threshold it will report a memory leak and fail the test.
```

> and monitoring V8 heap usage after every iteration. If it then detects that memory usage growth
> with a rate, higher then threshold it will report a memory leak and fail the test.

Reviewer 2: ```suggestion
and monitoring V8 heap usage after every iteration. If it detects that memory usage grows
with a rate higher than the threshold, it will report a memory leak and fail the test.
```
